gluten-ut/spark33/src/test/scala/org/apache/spark/sql/GlutenSQLQueryTestSuite.scala

```diff
@@ -768,35 +583,38 @@ class GlutenSQLQueryTestSuite
-            udf => UDFPgSQLTestCase(s"$testCaseName - ${udf.prettyName}", absPath, resultFile, udf)
-          }
-        } else if (file.getAbsolutePath.startsWith(s"$inputFilePath${File.separator}udf")) {
-          Seq(TestScalaUDF("udf"), TestPythonUDF("udf"), TestScalarPandasUDF("udf")).map {
-            udf => UDFTestCase(s"$testCaseName - ${udf.prettyName}", absPath, resultFile, udf)
-          }
-        } else if (file.getAbsolutePath.startsWith(s"$inputFilePath${File.separator}postgreSQL")) {
-          PgSQLTestCase(testCaseName, absPath, resultFile) :: Nil
-        } else if (file.getAbsolutePath.startsWith(s"$inputFilePath${File.separator}ansi")) {
-          AnsiTestCase(testCaseName, absPath, resultFile) :: Nil
-        } else if (
-          file.getAbsolutePath.startsWith(s"$inputFilePath${File.separator}timestampNTZ")
-        ) {
-          TimestampNTZTestCase(testCaseName, absPath, resultFile) :: Nil
-        } else {
-          RegularTestCase(testCaseName, absPath, resultFile) :: Nil
+    val createTestCase = (file: File, parentDir: String, resultPath: String) => {
+      val resultFile = file.getAbsolutePath.replace(parentDir, resultPath) + ".out"
+      val absPath = file.getAbsolutePath
+      val testCaseName = absPath.stripPrefix(parentDir).stripPrefix(File.separator)
+
+      if (
+        file.getAbsolutePath.startsWith(
+          s"$parentDir${File.separator}udf${File.separator}postgreSQL")
+      ) {
+        Seq(TestScalaUDF("udf"), TestPythonUDF("udf"), TestScalarPandasUDF("udf")).map {
+          udf => UDFPgSQLTestCase(s"$testCaseName - ${udf.prettyName}", absPath, resultFile, udf)
+        }
+      } else if (file.getAbsolutePath.startsWith(s"$parentDir${File.separator}udf")) {
+        Seq(TestScalaUDF("udf"), TestPythonUDF("udf"), TestScalarPandasUDF("udf")).map {
+          udf => UDFTestCase(s"$testCaseName - ${udf.prettyName}", absPath, resultFile, udf)
         }
+      } else if (file.getAbsolutePath.startsWith(s"$parentDir${File.separator}postgreSQL")) {
+        PgSQLTestCase(testCaseName, absPath, resultFile) :: Nil
+      } else if (file.getAbsolutePath.startsWith(s"$parentDir${File.separator}ansi")) {
+        AnsiTestCase(testCaseName, absPath, resultFile) :: Nil
+      } else if (file.getAbsolutePath.startsWith(s"$parentDir${File.separator}timestampNTZ")) {
+        TimestampNTZTestCase(testCaseName, absPath, resultFile) :: Nil
+      } else {
+        RegularTestCase(testCaseName, absPath, resultFile) :: Nil
+      }
     }
+    val overwriteTestCases = listFilesRecursively(new File(overwriteInputFilePath))
+      .flatMap(createTestCase(_, overwriteInputFilePath, overwriteGoldenFilePath))
+    val overwriteTestCaseNames = overwriteTestCases.map(_.name)
+    listFilesRecursively(new File(inputFilePath))
+      .flatMap(createTestCase(_, inputFilePath, goldenFilePath))
+      .filterNot(testCase => overwriteTestCaseNames.contains(testCase.name)) ++ overwriteTestCases
```

Comment:
What are the thoughts here to use contains instead of equals?

Reply:
overwriteTestCaseNames is `Seq[String]`, here it checks whether the `testCase.name`   is in the list.